lisa/tools/sysbench.py

```diff
@@ -0,0 +1,481 @@
+        valid_io_mode: List[str] = ["sync", "async", "mmap"]
+        valid_fsync_mode: List[str] = ["fsync", "fdatasync"]
+        valid_test_mode: Dict[str, str] = {
+            "seqwr": "write",
+            "seqrd": "read",
+            "rndrd": "read",
+            "rndwr": "write",
+            "seqrewr": "all",
+            "rndrw": "all",
+        }
+        perf_msg: Dict[Any, Any] = {}
+        testcase_name: str = inspect.stack()[1][3]
+        fsync_end: str = "on" if file_fsync_end else "off"
+
+        if operation not in valid_test_mode.keys():
+            raise LisaException(
+                f"Invalid test_mode. Valid test_mode: {valid_test_mode.keys()}"
+            )
+        elif file_io_mode not in valid_io_mode:
+            raise LisaException(
+                f"Invalid file_io_mode. Valid file_io_mode: {valid_io_mode}"
+            )
+        elif file_fsync_mode not in valid_fsync_mode:
+            raise LisaException(
+                f"Invalid file_fsync_mode. Valid file_fsync_mode: {valid_fsync_mode}"
+            )
+
+        extra_args: str = (
+            f" --file-test-mode={operation} --file-num={total_file}"
+            f" --file-block-size={block_size} --file-total-size={total_size}"
+            f" --file-io-mode={file_io_mode} --file-async-backlog={file_async_backlog}"
+            f" --file-fsync-freq={file_fsync_freq} --file-fsync-all={file_fsync_all}"
+            f" --file-fsync-end={fsync_end} --file-fsync-mode={file_fsync_mode}"
+            f" --file-merged-requests={file_merged_requests}"
+            f" --file-rw-ratio={file_rw_ratio}"
+        )
+
+        self.run(
+            f"fileio prepare --file-total-size={total_size} --file-num={total_file}",
+            force_run=True,
+        )
+
+        res = self.__run_sysbench_test(
+            test_type="fileio",
+            threads=threads,
+            events=events,
+            time_limit=time_limit,
+            validation="off",
+            extra_args=extra_args,
+        )
+
+        self.run(
+            f"fileio cleanup --file-total-size={total_size} --file-num={total_file}",
+            force_run=True,
+        )
+
+        perf_msg = self.__process_fileio_perf_result(
+            res.stdout,
+            valid_test_mode.get(operation, ""),
+        )
+        perf_msg["file_fsync_all"] = file_fsync_all
+        perf_msg["file_fsync_end"] = file_fsync_end
+        perf_msg["total_file"] = total_file
+        perf_msg["file_total_size_in_gb"] = file_total_size_in_gb
+        perf_msg["file_async_backlog"] = file_async_backlog
+        perf_msg["file_fsync_freq"] = file_fsync_freq
+        perf_msg["file_merged_requests"] = file_merged_requests
+        perf_msg["file_rw_ratio"] = file_rw_ratio
+        perf_msg["file_ops"] = operation
+        perf_msg["file_io_mode"] = file_io_mode
+        perf_msg["file_fsync_mode"] = file_fsync_mode
+        perf_msg["test_case_name"] = testcase_name
```

Comment:
The test_case_name is added below, so it doesn't need to be set in dict.

Reply:
Removed now